src/core/sandbox.c

```diff
@@ -0,0 +1,801 @@
+		O_RDONLY,
+		O_RDONLY|O_CLOEXEC,
+		O_RDONLY|O_NOCTTY|O_NONBLOCK,
+		O_WRONLY|O_CREAT|O_APPEND,
+		O_WRONLY|O_CREAT|O_TRUNC,
+		O_RDWR|O_CREAT|O_TRUNC
+	};
+	for (i = 0; i < 6; i++) {
+		rc = seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(open), 1,
+			SCMP_A1(SCMP_CMP_EQ, open_flag_array[i]));
+		if (rc < 0)
+			goto fail;
+	}
+
+	rc = seccomp_syscall_priority(ctx, SCMP_SYS(open), 60);
+	if (rc < 0)
+		goto fail;
+
+	/* openat */
+	rc = seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(openat), 2,
+		SCMP_A0(SCMP_CMP_EQ, AT_FDCWD),
+		SCMP_A2(SCMP_CMP_EQ, O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC));
+	if (rc < 0)
+		goto fail;
+
+	rc = seccomp_syscall_priority(ctx, SCMP_SYS(openat), 50);
+	if (rc < 0)
+		goto fail;
+
+	/* readlink */
+	rc = seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(readlink), 1,
+		SCMP_A2(SCMP_CMP_EQ, 4095));
+	if (rc < 0)
+		goto fail;
+
+	rc = seccomp_syscall_priority(ctx, SCMP_SYS(readlink), 50);
+	if (rc < 0)
+		goto fail;
+
+	/* getdents */
+	rc = seccomp_rule_add(ctx, SCMP_ACT_ERRNO(EBADF), SCMP_SYS(getdents), 1,
+		SCMP_A2(SCMP_CMP_EQ, 32768));
+	if (rc < 0)
+		goto fail;
+
+	rc = seccomp_syscall_priority(ctx, SCMP_SYS(getdents), 70);
+	if (rc < 0)
+		goto fail;
+
+	/* clock_gettime */
+	rc = seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(clock_gettime), 1,
+		SCMP_A0(SCMP_CMP_EQ, CLOCK_PROCESS_CPUTIME_ID));
+	if (rc < 0)
+		goto fail;
+
+	rc = seccomp_syscall_priority(ctx, SCMP_SYS(clock_gettime), 60);
+	if (rc < 0)
+		goto fail;
+
+	/* getrusage */
+	rc = seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(getrusage), 1,
+		SCMP_A0(SCMP_CMP_EQ, 0)); /* RUSAGE_SELF */
+	if (rc < 0)
+		goto fail;
+
+	rc = seccomp_syscall_priority(ctx, SCMP_SYS(getrusage), 50);
+	if (rc < 0)
+		goto fail;
+
+	/* select */
+	rc = seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(select), 2,
+		SCMP_A2(SCMP_CMP_EQ, NULL),
+		SCMP_A3(SCMP_CMP_EQ, NULL));
+	if (rc < 0)
+		goto fail;
+
+	rc = seccomp_syscall_priority(ctx, SCMP_SYS(poll), 200);
+	if (rc < 0)
+		goto fail;
+
+	/* poll */
+	rc = seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(poll), 2,
+		SCMP_A1(SCMP_CMP_LE, 3),
+		SCMP_A2(SCMP_CMP_GE, 0),
+		SCMP_A2(SCMP_CMP_LE, 1000));
+	if (rc < 0)
+		goto fail;
+
+	rc = seccomp_syscall_priority(ctx, SCMP_SYS(poll), 200);
+	if (rc < 0)
+		goto fail;
+
+	/* wait4 */
+	rc = seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(wait4), 2,
+		SCMP_A2(SCMP_CMP_EQ, WNOHANG),
+		SCMP_A3(SCMP_CMP_EQ, NULL));
+	if (rc < 0)
+		goto fail;
+
+	rc = seccomp_syscall_priority(ctx, SCMP_SYS(wait4), 50);
+	if (rc < 0)
+		goto fail;
+
+	/* tgkill */
+	rc = seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(tgkill), 2,
+		SCMP_A0(SCMP_CMP_EQ, getpid()),
+		SCMP_A2(SCMP_CMP_EQ, SIGABRT));
+	if (rc < 0)
+		goto fail;
+
+	rc = seccomp_syscall_priority(ctx, SCMP_SYS(tgkill), 50);
+	if (rc < 0)
+		goto fail;
+
+	/* apply filters */
+	rc = seccomp_load(ctx);
+	if (rc < 0)
+		goto fail;
+
+	/* seccomp has been seccessfully loaded and is now enforced */
+	return;
```

Comment:
you might want to `seccomp_release()` the filter after loading it as well, before this return call, in case the user cares about their memory load.